lib/adagrams.rb

```diff
@@ -0,0 +1,166 @@
+# Chantelle and Semret (Pair-Programing)
+# Adagram project
+# 8/13/18
+require 'csv'
+
+# WAVE 1 SYNTAX
+def draw_letters
+
+  random_draw = []
+
+  letter_pool = ('A'..'Z').map do |char|
+    case char
+    when 'A', 'I'
+      letter = char * 9
+    when 'B', 'C', 'F', 'H', 'M', 'P', 'V', 'W', 'Y'
+      letter = char * 2
+    when 'D', 'L', 'S', 'U'
+      letter = char * 4
+    when 'E'
+      letter = char * 12
+    when 'G'
+      letter = char * 3
+    when 'J', 'K', 'Q', 'X', 'Z'
+      letter = char * 1
+    when 'N', 'R', 'T'
+      letter = char * 6
+    when 'O'
+      letter = char * 8
+    end
+  end
+
+  split_letter_pool = letter_pool.map do |chars|
+    chars.split("")
+  end
+
+  #random draw of 10 from array and flatten the letter pool
+  random_draw = split_letter_pool.flatten.sample(10)
+  return random_draw
+
+end
+
+# puts "Random draw: #{draw_letters}"
+
+def uses_available_letters?(word, letter_in_hand)
+  temp_hand = letter_in_hand.dup
+  inputted_word = []
+  inputted_word << word.upcase.split("")
+  inputted_word.flatten!
+
+  check_letter = true
+
+  inputted_word.each do |letter|
+    index = 0
```

Comment:
Scope question: What is the smallest scope (the most specific/local block of code) that `index` needs to be in?